src/server/services/procedures/nexrad-radar/nexrad-radar.js

```diff
@@ -268,6 +268,17 @@ NexradRadar._addHurricane = async function(radar, plot, radarPlot, settings) {
     }
 };
 
+NexradRadar._filterRadars = async function(radars) {
+    for(let i = 0; i < radars.length; ++i) {
+        for(let j = i + 1; j < radars.length; ++j) {
+            if(this._getDistanceFromLatLonInKm(RADAR_LOCATIONS[radars[i]][0], RADAR_LOCATIONS[radars[i]][1], RADAR_LOCATIONS[radars[j]][0], RADAR_LOCATIONS[radars[j]][1]) <= RANGE) {
+                radars.splice(j, 1);
+            }
+        }
+    }
```

Comment:
When you remove item `j`, all the indices after that do down by `1`, but you `++j` regardless. So if `j=1`, you delete `1`, `2` becomes the new `1`, but you `++j` up to `2`, which skips over the original `2`. Modifying arrays during iteration is sometimes dangerous like that - this case can be fixed by iterating `j` backwards until it reaches `i`.

Reply:
Hi Devin. I just finished the revision and made a new commit. Now on my computer it runs 40s when the zoom level is 5 and 30s when the zoom level is 6.